test/gpu/fuse_mlir.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2022 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <migraphx/dead_code_elimination.hpp>
+#include <migraphx/gpu/fuse_mlir.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/pass_manager.hpp>
+#include <migraphx/program.hpp>
+#include <basic_ops.hpp>
+#include <migraphx/make_op.hpp>
+
+#include <test.hpp>
+#include <pointwise.hpp>
+
+void run_pass(migraphx::program& p)
+{
+    migraphx::run_passes(p, {migraphx::gpu::fuse_mlir{}, migraphx::dead_code_elimination{}});
+}
+
+// cppcheck-suppress definePrefix
+// NOLINTNEXTLINE
+#define VERIFY_MLIR(pass, ref)            \
+    do                                    \
+    {                                     \
+        if(migraphx::gpu::mlir_enabled()) \
```

Comment:
Yea I dont think you can check this here, unless we want to move the `mlir_op` outside of the ifdef.

Reply:
Isn't `mlir_enabled` present even when MLIR is off and then it just statically returns false?

Reply:
Yes, but the problem is that there is no `mlir_op` so constructing the test fails when `mlir_enabled` returns false. You can either move the condition into the test case at the beginning or move the `mlir_op` out of the `ifdef`.

This is why its failing on the CI.

Reply:
Ah.

How do y'all generally handle this sort of thing? Are other conditionally-enableable ops defined outside off the relevant ifdef?

I might end up moving the enable_mlir to main

Reply:
> I might end up moving the enable_mlir to main

Thats seems like a simpler solution.